File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/ios/Classes/Presentation/{{name.pascalCase()}}PlatformViewFactory.swift
// Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.

// coverage:ignore-file

import Flutter
import UIKit

public class {{name.pascalCase()}}PlatformViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger?
    private var viewModel: {{name.pascalCase()}}ViewModel?

    public init(messenger: FlutterBinaryMessenger? = nil, viewModel: {{name.pascalCase()}}ViewModel? = nil) {
        self.messenger = messenger
        self.viewModel = viewModel
        super.init()
    }

    public func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        return {{name.pascalCase()}}PlatformView(
            frame: frame,
            viewIdentifier: viewId,
            arguments: args,
            binaryMessenger: messenger,
            viewModel: viewModel
        )
    }

    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}

File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/android/src/main/kotlin/com/danhdue/{{name.snakeCase()}}/platform/{{name.pascalCase()}}PlatformViewFactory.kt
/*
 * Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.
 */

/*
 * coverage:ignore-file
 */

package com.danhdue.{{name.snakeCase()}}.platform

import android.content.Context
import com.danhdue.{{name.snakeCase()}}.presentation.{{name.pascalCase()}}PlatformView
import com.danhdue.{{name.snakeCase()}}.presentation.{{name.pascalCase()}}ViewModel
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class {{name.pascalCase()}}PlatformViewFactory(
    private val viewModelProvider: () -> {{name.pascalCase()}}ViewModel
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    constructor(viewModel: {{name.pascalCase()}}ViewModel) : this({ viewModel })

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        return {{name.pascalCase()}}PlatformView(context, viewId, args, viewModelProvider())
    }
}
